frontend/src/app/chatmenu/BlockUser.tsx

```diff
@@ -1,66 +1,79 @@
-import React from 'react'
+import React, { useState } from 'react'
 import './BlockUser.css'
-import '../ui/User.css'
 
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { faUnlock, faUserLock } from '@fortawesome/free-solid-svg-icons'
+import { faLock, faLockOpen } from '@fortawesome/free-solid-svg-icons'
 
 import { ChatMenuPageProps } from './ChatMenuPage'
 import { UserData } from '../types/UserData'
+import { Button, UserTile } from '../ui'
+import { fetchFromBackend, useBackendFetchJson } from '../utils'
 
 interface BlockUserProps {
     participants: UserData[]
+    myUid: string
+}
+
+function FetchBlockedUsers(myUid: string, blockUid: string) {
+    const [res] = useBackendFetchJson({
+        route: `/chat/block/user/${myUid}`,
+        deps: [myUid],
+        isValid: myUid,
+    })
+    const blockedUsers = res?.blocks || []
+
+    return blockedUsers.find((user: { uid: string }) => user?.uid === blockUid)?.blocked || false
```

Comment:
Instead of using `find()` here, we could reduce the list into a dictionary of `uid` to `blocked`, then when we loop over the participants, we can pull the blocked status from the dictionary in constant time, with `false` as the fallback value if the `uid` is not found.

There are only a maximum of three users in participants, but there could be more users in the blocked list. Even though the total number is not likely to be high enough to cause a performance issue, we can still improve the runtime and demonstrate good frontend performance for an efficient user interface.